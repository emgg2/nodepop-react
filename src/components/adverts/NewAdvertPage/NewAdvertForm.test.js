import React from 'react';
import { shallow } from 'enzyme';

import NewAdvertForm from './NewAdvertForm';

describe('NEW ADVERT FORM', () => {
    const props = {
        onSubmit: jest.fn()
    }

    const render = () => shallow(<NewAdvertForm {...props} />);

    test('should render', () => {
        const wrapper = render ();
        expect(wrapper.exists()).toBe(true);
    })

    test('should call submit', () => {
        const wrapper = render();
        const form = wrapper.find('form');
        form.simulate('submit', {preventDefault: () => {}});
        expect(props.onSubmit).toHaveBeenCalled();
    })

    test('should submit advert', () => {
        const advert = {
            name: 'Prod1',
            sale: true,
            price: 20, 
            tags: ['tag1'], 
            photo: null
        }
        const wrapper = render();
        const nameField = wrapper.find('[name="name"]');
        nameField
            .props()
            .onChange({target: {name: 'name', value: advert.name}});

        const saleField = wrapper.find('[name="sale"]');
        saleField
            .props()
            .onChange({target: {name: 'sale', value: advert.sale}});

        const priceField = wrapper.find('[name="price"]');
        priceField
            .props()
            .onChange({target: {name: 'price', value: advert.price}});
        
        const tagsField = wrapper.find('[name="tags"]');
        tagsField
            .props()
            .onChange({target: {name: 'tags', value: advert.tags}});
        
        const photoField = wrapper.find('[name="photo"]');
        photoField
            .props()
            .onChange({target: {name: 'photo', value: advert.photo}});
    
        const form = wrapper.find ('form');
        form.simulate('submit', { preventDefault: () => {}});
        expect(wrapper.find('button').props().disabled).toBe(false);
        expect(props.onSubmit).toHaveBeenCalledWith(advert);       
        
    }); 
    
    test('snapshot testing', () => {
        const wrapper = render();
        expect (wrapper).toMatchSnapshot();
    })

});