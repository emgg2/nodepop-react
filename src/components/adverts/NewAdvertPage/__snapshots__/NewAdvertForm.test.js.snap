// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW ADVERT FORM snapshot testing 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="name"
    onChange={[Function]}
    value=""
  />
  <input
    checked={true}
    name="sale"
    onChange={[Function]}
    type="checkbox"
  />
  <input
    name="price"
    onChange={[Function]}
    type="number"
    value={0}
  />
  <SelectTags
    multiple={true}
    name="tags"
    onChange={[Function]}
    value={Array []}
  />
  <InputFile
    name="photo"
    onChange={[Function]}
  />
  <button
    disabled={true}
  >
    Save
  </button>
</form>
`;

exports[`NEW ADVERT FORM snapshot testing with button disabled 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="name"
    onChange={[Function]}
    value="Prod1"
  />
  <input
    checked={true}
    name="sale"
    onChange={[Function]}
    type="checkbox"
  />
  <input
    name="price"
    onChange={[Function]}
    type="number"
    value={20}
  />
  <SelectTags
    multiple={true}
    name="tags"
    onChange={[Function]}
    value={
      Array [
        "tag1",
      ]
    }
  />
  <InputFile
    name="photo"
    onChange={[Function]}
  />
  <button
    disabled={false}
  >
    Save
  </button>
</form>
`;
